docroot/modules/custom/mass_site_map/src/Plugin/simple_sitemap/UrlGenerator/DocumentPagesUrlGenerator.php

```diff
@@ -40,11 +40,11 @@ public function processDataSet($entity) {
     foreach ($files as $file) {
```

Comment:
Are all $files always media entities? If not, we will be generating invalid links?

Reply:
We are not generating invalid links, because we are generating URLs not from the $file but from the host entity which is always Media entity type. I will add some checks to avoid issues in the future

Reply:
Added checks.